<robot name="robot_hanchenhao">

  <link name="base_link">
    <visual>
      <geometry>
        <box size="0.4 0.8 0.2"/>
      </geometry>
      <material name="gray">
        <color rgba=".2 .2 .2 1"/>
      </material>
    </visual>
  </link>

  <link name="body_link"> 
    <visual>
      <geometry>
        <cylinder length="1.8" radius="0.36"/>
      </geometry>
      <material name="red">
        <color rgba="1 0 0 1"/>
      </material>
    </visual>
  </link>

  <link name="wheel_left_link">
    <visual>
      <origin xyz="0 0 0" rpy="1.57 0 0" />
      <geometry>
        <cylinder length="0.2" radius="0.35"/>
      </geometry>
      <material name="black">
        <color rgba="0 0 0 1"/>
      </material>
    </visual>
  </link>

  <link name="wheel_right_link" >
    <visual>
      <origin xyz="0 0 0" rpy="1.57 0 0" />
      <geometry>
        <cylinder length="0.2" radius="0.35"/>
      </geometry>
      <material name="black">
        <color rgba="0 0 0 1"/>
      </material>
    </visual>
  </link>

  <link name="hand_left_link" >
    <visual>
      <origin xyz="0.25 -0.1 0" rpy="0 1.57 0" />
      <geometry>
        <box size="0.1 0.2 0.6"/>
      </geometry>
      <material name="yellow">
        <color rgba="1 1 0 1"/>
      </material>
    </visual>
  </link>

  <link name="hand_right_link">
    <visual>
      <origin xyz="0.25 0.1 0" rpy="0 1.57 0" />
      <geometry>
        <box size="0.1 0.2 0.6"/>
      </geometry>
      <material name="yellow">
        <color rgba="1 1 0 1"/>
      </material>
    </visual>
  </link>

  <link name="head_link">
    <visual>
      <geometry>
        <sphere radius="0.4"/>
      </geometry>
      <material name="yellow">
        <color rgba="1 1 0 1"/>
      </material>
    </visual>
  </link>

  <joint name="joint1" type="fixed">
    <parent link="base_link"/>
    <child link="wheel_left_link"/>
    <origin xyz="0 0.5 0" rpy="0 0 0" />
  </joint>

  <joint name="joint2" type="fixed">
    <parent link="base_link"/>
    <child link="wheel_right_link"/>
    <origin xyz="0 -0.5 0" rpy="0 0 0" />
  </joint>

  <joint name="joint3" type="fixed">
    <parent link="base_link"/>
    <child link="body_link"/>
    <origin xyz="0 0 1" rpy="0 0 0" />
  </joint>

  <joint name="joint4" type="continuous">
    <parent link="body_link"/>
    <child link="hand_left_link"/>
    <origin xyz="0 0.5 0.2" rpy="0 0 0" />
    <axis xyz="0 1 0"/>
  </joint>

  <joint name="joint5" type="continuous">
    <parent link="body_link"/>
    <child link="hand_right_link"/>
    <origin xyz="0 -0.5 0.2" rpy="0 0 0" />
    <axis xyz="0 -1 0"/>
  </joint>

  <joint name="joint6" type="prismatic">
    <parent link="body_link"/>
    <child link="head_link"/>
    <limit effort="10" lower="-0.5" upper="0.5" velocity="10"/>
    <origin xyz="0 0 1" rpy="0 0 0" />
    <axis xyz="0 0 1"/>
  </joint>

</robot>
